### PR TITLE
Fix testing and add compilation test.

### PR DESCRIPTION
**Pull request by chriseth**
This removes some unnecessary timeouts from tests and especially adds the first test that compiles a source file.

Lessons learnt from these tests:
- It takes more than 30 seconds to load the compiler in the test environment

Because of that, I would propose to put as many tests as possible into single test cases that only need to load the compiler once.

### CODE REVIEW

**Comment by chriseth** [comment]
This is still not too stable. I fear that there is no way around mocking the compiler. One way to do that is to provide a mock file `soljson.js` in the local repository at the time the tests are run and then use the url `http://127.0.0.1:8080/#version=builtin` for the tests.

In order for this to run smoothly, the tests could provide a list of anything they will compile and then travis compiles these sources before the UI tests are run and puts these compilations results inside the mock soljson.js.

File: ci/browser_tests.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
-
-SAUCECONNECT_URL="https://saucelabs.com/downloads/sc-4.3.16-linux.tar.gz"
+SC_VERSION="4.4.0"
```
Comment (axic) [inline ci/browser_tests.sh]: Can you split out the changes to `browser_tests.sh` into another PR and merge that?

File: nightwatch.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ module.exports = {
 
     'local': {
       'launch_url': 'http://localhost:8080',
-      'selenium_port': 4444,
+      'selenium_port': 4445,
```
Comment (chriseth) [inline nightwatch.js]: Weird - but it seems to work, perhaps that option is ignored.

File: ci/browser_tests.sh (hunk 1 above)
Comment (chriseth) [inline ci/browser_tests.sh]: Will do, the actual tests will probably need more change as mentioned above.

**Comment by axic** [comment]
Is this obsoleted by #273?

**Comment by axic** [comment]
Rebased.

**Comment by chriseth** [comment]
I will update this once https://github.com/ethereum/browser-solidity/pull/283 is merged.

File: test-browser/compiling.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+module.exports = {
+  'Compile Simple Contract': function (browser) {
+    browser
+      .url('http://127.0.0.1:8080')
```
Comment (axic) [inline test-browser/compiling.js]: Should this load a specific version? Like 0.4.0?

**Comment by yann300** [comment]
this PR also fix an ie/safari bug
throw "attempted to assign to readonly property exception" here:
https://github.com/ethereum/browser-solidity/pull/250/files#diff-c2ddeee397dd6378fa147d13ec494e31R830

File: src/universal-dapp.js [hunk 4]
```diff
@@ -826,8 +827,11 @@ UniversalDApp.prototype.rawRunTx = function (args, cb) {
         ++self.blockNumber
       }
       self.vm.runTx({block: block, tx: tx, skipBalance: true, skipNonce: true}, function (err, result) {
-        result.transactionHash = self.txdebugger.web3().releaseCurrentHash() // used to keep track of the transaction
```
Comment (axic) [inline src/universal-dapp.js]: I sitll don't get where is `result` readonly? Safari just takes any input parameter as readonly?

Comment (axic) [inline src/universal-dapp.js]: Also the web3 mode must be updated with this change.

File: test-browser/tests/new-file-test.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ module.exports = {
     browser
       .waitForElementVisible('.newFile', 10000)
       .click('.newFile')
-      .pause('10000')
+      .pause(500)
```
Comment (axic) [inline test-browser/tests/new-file-test.js]: Is the new value better / enough?

File: test-browser/tests/smoke-test.js [hunk 6]
```diff
@@ -10,7 +10,6 @@ module.exports = {
   'Smoke test': function (browser) {
     browser
       .waitForElementVisible('#righthand-panel', 10000)
-      .pause('10000')
```
Comment (axic) [inline test-browser/tests/smoke-test.js]: Is this pause not required anymore?

**Comment by axic** [comment]
Moved the readonly fix to #305.

File: test-browser/tests/smoke-test.js (hunk 6 above)
Comment (yann300) [inline test-browser/tests/smoke-test.js]: no, we wait just before

**Pull request merged**